jolt-evm-verifier/script/src/bin/hyperkzg_batch_example.rs

```diff
@@ -0,0 +1,173 @@
+    let (pk, vk): (HyperKZGProverKey<Bn254>, HyperKZGVerifierKey<Bn254>) = srs.trim(n);
+
+    let point = (0..ell)
+        .map(|_| <Bn254 as Pairing>::ScalarField::rand(&mut rng))
+        .collect::<Vec<_>>();
+
+    let mut polys = vec![];
+    let mut evals = vec![];
+    let mut commitments = vec![];
+    let mut borrowed = vec![];
+    for _ in 0..8 {
+        let poly = DensePolynomial::new(
+            (0..n)
+                .map(|_| <Bn254 as Pairing>::ScalarField::rand(&mut rng))
+                .collect::<Vec<_>>(),
+        );
+        let eval = poly.evaluate(&point);
+        commitments.push(HyperKZG::commit(&pk, &poly).unwrap());
+        polys.push(poly);
+        evals.push(eval);
+    }
+
+    for poly in polys.iter() {
+        borrowed.push(poly);
+    }
+
+    // prove an evaluation
+    let mut prover_transcript = ProofTranscript::new(b"TestEval");
+    let proof: HyperKZGProof<Bn254> = HyperKZG::batch_prove(
+        &(pk, vk),
+        borrowed.as_slice(),
+        &point,
+        &evals,
+        BatchType::Big,
+        &mut prover_transcript,
+    );
+
+    sol!(struct VK {
+        uint256 VK_g1_x;
+        uint256 VK_g1_y;
+        uint256[] VK_g2;
+        uint256[] VK_beta_g2;
+    });
+    sol!(struct HyperKZGProofSol {
+        uint256[] com; // G1 points represented pairwise
+        uint256[] w; // G1 points represented pairwise
+        uint256[] v_ypos; // Three vectors of scalars which must be ell length
+        uint256[] v_yneg;
+        uint256[] v_y;
+    });
+    sol!(struct BatchedExample {
+        VK vk;
+        HyperKZGProofSol proof;
+        uint256[] commitments;
+        uint256[] point;
+        uint256[] claims;
+    });
+
+    // We must invert the vk point on g2
+    let g1 = vk.kzg_vk.g1;
+    let g2 = -vk.kzg_vk.g2;
+    let g2_sol = vec![
+        U256::from_be_slice(&g2.x.c0.into_bigint().to_bytes_be()),
+        U256::from_be_slice(&g2.x.c1.into_bigint().to_bytes_be()),
+        U256::from_be_slice(&g2.y.c0.into_bigint().to_bytes_be()),
+        U256::from_be_slice(&g2.y.c1.into_bigint().to_bytes_be()),
+    ];
+    let g2_beta = vk.kzg_vk.beta_g2;
+    let g2_beta_sol = vec![
+        U256::from_be_slice(&g2_beta.x.c0.into_bigint().to_bytes_be()),
+        U256::from_be_slice(&g2_beta.x.c1.into_bigint().to_bytes_be()),
+        U256::from_be_slice(&g2_beta.y.c0.into_bigint().to_bytes_be()),
+        U256::from_be_slice(&g2_beta.y.c1.into_bigint().to_bytes_be()),
+    ];
+
+    let vk_sol = VK {
+        VK_g1_x: U256::from_be_slice(&g1.x.into_bigint().to_bytes_be()),
+        VK_g1_y: U256::from_be_slice(&g1.y.into_bigint().to_bytes_be()),
+        VK_g2: g2_sol,
+        VK_beta_g2: g2_beta_sol,
+    };
+
+    let mut com = vec![];
+    let mut w = vec![];
+    let ypos_scalar = proof.v[0].clone();
+    let yneg_scalar = proof.v[1].clone();
+    let y_scalar = proof.v[2].clone();
+
+    // Horrible type conversion here, possibly theres an easier way
```

Comment:
Unfortunately the Bn254 points are stored in Montgomery form so they incur a field multiplication to get to BigInt form. I don't know how the EVM precompiles handle this but its possible we want to keep them in Montgomery form the whole time.

Reply:
No sadly it assumes a affine non mongomery form